# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux/hotfix-amd-cppc.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-6.4/drivers/cpufreq/amd-pstate.c.vanilla	2023-08-25 18:17:28.975750904 +0200
+++ linux-6.4/drivers/cpufreq/amd-pstate.c	2023-08-25 18:19:33.680747480 +0200
@@ -281,7 +281,7 @@
 	 * calculated wrongly. we take the AMD_CPPC_HIGHEST_PERF(cap1) value as
 	 * the default max perf.
 	 */
-	if (prefcore)
+	if (!prefcore)
 		WRITE_ONCE(cpudata->highest_perf, AMD_PSTATE_PREFCORE_THRESHOLD);
 	else
 		WRITE_ONCE(cpudata->highest_perf, AMD_CPPC_HIGHEST_PERF(cap1));
@@ -302,7 +302,7 @@
 	if (ret)
 		return ret;
 
-	if (prefcore)
+	if (!prefcore)
 		WRITE_ONCE(cpudata->highest_perf, AMD_PSTATE_PREFCORE_THRESHOLD);
 	else
 		WRITE_ONCE(cpudata->highest_perf, cppc_perf.highest_perf);
